src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -50,6 +59,35 @@ object EntityClient {
     }
   }
 
+  //returns (contents of participants.tsv, contents of samples.tsv)
+  def unzipTSVs(zipFile: ZipFile): (Option[String], Option[String]) = {
+    val zipEntries = zipFile.entries.asScala
+
+    val rand = java.util.UUID.randomUUID.toString.take(8)
+    val participantsFn = s"/tmp/$rand-participants.tsv"
+    val samplesFn = s"/tmp/$rand-samples.tsv"
+
+    val unzippedFiles = zipEntries.foldLeft((None: Option[String], None: Option[String])){ (acc: (Option[String], Option[String]), ent: ZipEntry) =>
+      if(!ent.isDirectory && ent.getName.contains("participants.tsv")) {
```

Comment:
If there are multiple matching files in different directories, this will pick the "last" one based on whatever order is returned from the `ZipFile`. I'd rather we either explicitly search only the root (or wherever we agree they will live) or fail if there isn't a standard location and we find more than one match for either filename.

(I skipped over this when looking at the preview because of the `//TODO: eww` comment.)

Reply:
fair point. i'll accumulate all of them and if there's multiples i'll throw an annoyed error